src/main/scala/com/lucidworks/spark/util/SolrQuerySupport.scala

```diff
@@ -0,0 +1,653 @@
+          val name = payload.get("name").get.asInstanceOf[String]
+          val fieldType = payload.get("type").get.asInstanceOf[String]
+
+          val isRequired: Option[Boolean] = {
+            if (payload.contains("required")) {
+              if (payload.get("required").isDefined) {
+                payload.get("required").get match {
+                  case v: Boolean => Some(v)
+                  case v1: AnyRef => Some(String.valueOf(v1).equals("true"))
+                }
+              } else None
+            } else None
+          }
+
+          val isMultiValued: Option[Boolean] = {
+            if (payload.contains("multiValued")) {
+              if (payload.get("multiValued").isDefined) {
+                payload.get("multiValued").get match {
+                  case v: Boolean => Some(v)
+                  case v1: AnyRef => Some(String.valueOf(v1).equals("true"))
+                }
+              } else None
+            } else None
+          }
+
+          val isStored: Option[Boolean] = {
+            if (payload.contains("stored")) {
+              if (payload.get("stored").isDefined) {
+                payload.get("stored").get match {
+                  case v: Boolean => Some(v)
+                  case v1: AnyRef => Some(String.valueOf(v1).equals("true"))
+                }
+              } else None
+            } else None
+          }
+
+          val isDocValues: Option[Boolean] = {
+            if (payload.contains("docValues")) {
+              if (payload.get("docValues").isDefined) {
+                payload.get("docValues").get match {
+                  case v: Boolean => Some(v)
+                  case v1: AnyRef => Some(String.valueOf(v1).equals("true"))
+                }
+              } else None
+            } else None
+          }
+
+          val dynamicBase: Option[String] = {
+            if (payload.contains("dynamicBase")) {
+              if (payload.get("dynamicBase").isDefined) {
+                payload.get("dynamicBase").get match {
+                  case v: String => Some(v)
+                }
+              } else None
+            } else None
+          }
+
+          val fieldClassType: Option[String] = {
+            if (fieldTypeToClassMap.contains(fieldType)) {
+              if (fieldTypeToClassMap.get(fieldType).isDefined) {
+                Some(fieldTypeToClassMap.get(fieldType).get)
+              } else None
+            } else None
+          }
+
+          val solrFieldMeta = SolrFieldMeta(fieldType, dynamicBase, isRequired, isMultiValued, isDocValues, isStored, fieldClassType)
+
+          if ((solrFieldMeta.isStored.isDefined && !solrFieldMeta.isStored.get) &&
+            (solrFieldMeta.isDocValues.isDefined && !solrFieldMeta.isDocValues.get)) {
+              log.debug("Can't retrieve an index only field: '" + name + "'. Field info " + payload)
+          } else if ((solrFieldMeta.isStored.isDefined && solrFieldMeta.isStored.get) &&
```

Comment:
Needs a NOT (!) on the isStored flag ;-) I have a fix in my code that I'll push soon